edx_rest_framework_extensions/auth/jwt/decoder.py

```diff
@@ -173,18 +173,73 @@ def _set_filters(token):
 
 
 def _verify_jwt_signature(token, jwt_issuer, decode_symmetric_token):
+    """
+    Verifies the JWT signature. Raises InvalidTokenError in the event of an error.
+
+    Arguments:
+        token (str): JWT to be decoded.
+        jwt_issuer (dict): A dict of JWT issuer related settings, containing the symmetric key.
+        decode_symmetric_token (bool): Whether to decode symmetric tokens or not. Pass False for asymmetric tokens only
+    """
+    # .. custom_attribute_name: jwt_auth_decode_symmetric_token
+    # .. custom_attribute_description: True if symmetric keys will also be used for checking
+    #   the JWT signature, and False if only asymmetric keys will be used.
+    set_custom_attribute('jwt_auth_decode_symmetric_token', decode_symmetric_token)
+
+    # For observability purposes, we will first try asymmetric keys only to verify
+    #   that we no longer need the symmetric key. However, if this fails, we will
+    #   continue on to the original code path and try all keys (including symmetric)
+    #   and add monitoring to let us know. This is meant to be temporary, until we
+    #   can fully retire code paths for symmetric keys, as part of
+    #   DEPR: Symmetric JWTs: https://github.com/openedx/public-engineering/issues/83
+
+    # Use add_symmetric_keys=False to only include asymmetric keys at first
+    key_set = _get_signing_jwk_key_set(jwt_issuer, add_symmetric_keys=False)
+    # .. custom_attribute_name: jwt_auth_verify_asymmetric_keys_count
+    # .. custom_attribute_description: Number of JWT verification keys in use for this
+    #   verification. Should be same as number of asymmetric public keys. This is
+    #   intended to aid in key rotations; once the average count stabilizes at a
+    #   higher number after adding a public key, it should be safe to change the secret key.
+    set_custom_attribute('jwt_auth_verify_asymmetric_keys_count', len(key_set))
+
+    try:
+        _ = JWS().verify_compact(token, key_set)
+        # .. custom_attribute_name: jwt_auth_asymmetric_verified
+        # .. custom_attribute_description: The count of JWTs successfully verified
+        #   using an asymmetric key.
```

Comment:
nit: We'll use this as a count, but for each request it's just a boolean.
```suggestion
        # .. custom_attribute_description: Whether the JWT was successfully verified
        #   using an asymmetric key.
```